### PR TITLE
Remove html names for docker repositories

### LINKED ISSUE #5593: Docker Repo tests - using HTML values in positive names
```
2017-11-04 07:29:47 - nailgun.client - DEBUG - Making HTTP POST request to https://sat6.com/katello/api/v2/repositories with options {'verify': False, 'auth': ('admin', 'changeme'), 'headers': {'content-type': 'application/json'}}, no params and data {"product_id": 1864, "url": "https://registry-1.docker.io", "docker_upstream_name": "busybox", "content_type": "docker", "name": "<q>kHaXiGqI</q>"}.
2017-11-04 07:29:47 - nailgun.client - WARNING - Received HTTP 500 response: {"displayMessage":"A validation error occurred.","errors":["A validation error occurred."]}
```

### PR DESCRIPTION
**Pull request by odovzhenko**
Close #5593

```
(sat-6.3.0) odovz@bueno:~/projects/robottelo$ pytest -v tests/foreman/api/test_docker.py::DockerRepositoryTestCase::test_positive_create_with_name tests/foreman/api/test_docker.py::DockerRepositoryTestCase::test_positive_update_name tests/foreman/cli/test_docker.py::DockerRepositoryTestCase::test_positive_create_with_name tests/foreman/cli/test_docker.py::DockerRepositoryTestCase::test_positive_update_name tests/foreman/ui/test_docker.py::DockerRepositoryTestCase::test_positive_create_with_name tests/foreman/ui/test_docker.py::DockerRepositoryTestCase::test_positive_update_name tests/foreman/ui/test_docker.py::DockerRepositoryTestCase::test_positive_delete tests/foreman/ui/test_docker.py::DockerContentViewTestCase::test_positive_publish_with_docker_repo
=========================================================================================== test session starts ============================================================================================
platform linux2 -- Python 2.7.12, pytest-3.2.3, py-1.4.34, pluggy-0.4.0 -- /home/odovz/venv/sat-6.3.0/bin/python2.7
cachedir: .cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/odovz/projects/robottelo, inifile:
plugins: xdist-1.20.1, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collected 8 items                                                                                                                                                                                           
2017-12-27 16:09:43 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/api/test_docker.py::DockerRepositoryTestCase::test_positive_create_with_name <- robottelo/decorators/__init__.py PASSED
tests/foreman/api/test_docker.py::DockerRepositoryTestCase::test_positive_update_name <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_docker.py::DockerRepositoryTestCase::test_positive_create_with_name <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_docker.py::DockerRepositoryTestCase::test_positive_update_name <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_docker.py::DockerRepositoryTestCase::test_positive_create_with_name <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_docker.py::DockerRepositoryTestCase::test_positive_update_name <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_docker.py::DockerRepositoryTestCase::test_positive_delete <- robottelo/decorators/__init__.py FAILED
tests/foreman/ui/test_docker.py::DockerContentViewTestCase::test_positive_publish_with_docker_repo <- robottelo/decorators/__init__.py FAILED

================================================================================================= FAILURES =================================================================================================
2017-12-27 17:30:11 - robottelo.ui.session - ERROR - Message: The test with session id e7abcc927a024ec7865ad0f365f5d7d4 has already finished, and can't receive further commands.
You can learn more at https://saucelabs.com/jobs/e7abcc927a024ec7865ad0f365f5d7d4
2017-12-27 17:30:11 - robottelo.test - DEBUG - Updating SauceLabs job "e7abcc927a024ec7865ad0f365f5d7d4": name "test_positive_delete (tests.foreman.ui.test_docker.DockerRepositoryTestCase)" and status "failed"


2017-12-27 17:59:15 - robottelo.ui.session - ERROR - Message: The test with session id dc62c869b32d46b1bdcd442d11ce58ef has already finished, and can't receive further commands.
You can learn more at https://saucelabs.com/jobs/dc62c869b32d46b1bdcd442d11ce58ef
2017-12-27 17:59:16 - robottelo.test - DEBUG - Updating SauceLabs job "dc62c869b32d46b1bdcd442d11ce58ef": name "test_positive_publish_with_docker_repo (tests.foreman.ui.test_docker.DockerContentViewTestCase)" and status "failed"

================================================================================== 2 failed, 6 passed in 6574.95 seconds ===================================================================================
```

```
(sat-6.3.0) odovz@bueno:~/projects/robottelo$ pytest -v tests/foreman/ui/test_docker.py::DockerRepositoryTestCase::test_positive_delete
=========================================================================================== test session starts ============================================================================================
platform linux2 -- Python 2.7.12, pytest-3.2.3, py-1.4.34, pluggy-0.4.0 -- /home/odovz/venv/sat-6.3.0/bin/python2.7
cachedir: .cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/odovz/projects/robottelo, inifile:
plugins: xdist-1.20.1, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collected 1 item                                                                                                                                                                                            
2017-12-27 18:42:59 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/ui/test_docker.py::DockerRepositoryTestCase::test_positive_delete <- robottelo/decorators/__init__.py PASSED

======================================================================================== 1 passed in 733.69 seconds ========================================================================================
```

```
(sat-6.3.0) odovz@bueno:~/projects/robottelo$ pytest -v tests/foreman/ui/test_docker.py::DockerContentViewTestCase::test_positive_publish_with_docker_repo
=========================================================================================== test session starts ============================================================================================
platform linux2 -- Python 2.7.12, pytest-3.2.3, py-1.4.34, pluggy-0.4.0 -- /home/odovz/venv/sat-6.3.0/bin/python2.7
cachedir: .cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/odovz/projects/robottelo, inifile:
plugins: xdist-1.20.1, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collected 1 item                                                                                                                                                                                            
2017-12-27 18:43:04 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/ui/test_docker.py::DockerContentViewTestCase::test_positive_publish_with_docker_repo <- robottelo/decorators/__init__.py PASSED

======================================================================================== 1 passed in 554.84 seconds ========================================================================================
```

### CODE REVIEW

File: robottelo/datafactory.py [hunk 1]
```diff
@@ -222,6 +222,21 @@ def valid_data_list():
...
+def valid_docker_repository_names():
+    """Generates a list of valid names for Docker repository."""
+    return [
+        gen_string('alphanumeric', random.randint(1, 255)),
+        gen_string('alpha', random.randint(1, 255)),
+        gen_string('cjk', random.randint(1, 85)),
+        gen_string('latin1', random.randint(1, 255)),
+        gen_string('numeric', random.randint(1, 255)),
+        gen_string('utf8', random.randint(1, 85)),
+        gen_string('html', random.randint(1, 85)) if not bz_bug_is_open(
```
Comment (oshtaier) [inline robottelo/datafactory.py]: I think that we should just skip that data input in case BZ is open, without else

**Comment by oshtaier** [comment]
tests should pass right away, without re-run. Appreciate if you can look on that more

**Comment by odovzhenko** [comment]
Previously tests failed due to 30 min duration timeout in saucelabs.

```(sat-6.3.0) odovz@bueno:~/projects/robottelo$ pytest -v tests/foreman/api/test_docker.py::DockerRepositoryTestCase::test_positive_create_with_name tests/foreman/api/test_docker.py::DockerRepositoryTestCase::test_positive_update_name tests/foreman/cli/test_docker.py::DockerRepositoryTestCase::test_positive_create_with_name tests/foreman/cli/test_docker.py::DockerRepositoryTestCase::test_positive_update_name tests/foreman/ui/test_docker.py::DockerRepositoryTestCase::test_positive_create_with_name tests/foreman/ui/test_docker.py::DockerRepositoryTestCase::test_positive_update_name tests/foreman/ui/test_docker.py::DockerRepositoryTestCase::test_positive_delete tests/foreman/ui/test_docker.py::DockerContentViewTestCase::test_positive_publish_with_docker_repo
=========================================================================================== test session starts ============================================================================================
platform linux2 -- Python 2.7.12, pytest-3.2.3, py-1.4.34, pluggy-0.4.0 -- /home/odovz/venv/sat-6.3.0/bin/python2.7
cachedir: .cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/odovz/projects/robottelo, inifile:
plugins: xdist-1.20.1, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collected 8 items                                                                                                                                                                                           
2017-12-28 12:44:22 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/api/test_docker.py::DockerRepositoryTestCase::test_positive_create_with_name <- robottelo/decorators/__init__.py PASSED
tests/foreman/api/test_docker.py::DockerRepositoryTestCase::test_positive_update_name <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_docker.py::DockerRepositoryTestCase::test_positive_create_with_name <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_docker.py::DockerRepositoryTestCase::test_positive_update_name <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_docker.py::DockerRepositoryTestCase::test_positive_create_with_name <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_docker.py::DockerRepositoryTestCase::test_positive_update_name <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_docker.py::DockerRepositoryTestCase::test_positive_delete <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_docker.py::DockerContentViewTestCase::test_positive_publish_with_docker_repo <- robottelo/decorators/__init__.py PASSED

======================================================================================= 8 passed in 1859.53 seconds ========================================================================================
```

File: robottelo/datafactory.py [hunk 2]
```diff
@@ -222,6 +222,21 @@ def valid_data_list():
     ]
 
 
+@filtered_datapoint
+def valid_docker_repository_names():
+    """Generates a list of valid names for Docker repository."""
+    names = [gen_string('alphanumeric', random.randint(1, 255)),
```
Comment (oshtaier) [inline robottelo/datafactory.py]:
```
      names = [
            gen_string('alphanumeric', random.randint(1, 255)),
            gen_string('alpha', random.randint(1, 255)),
            gen_string('cjk', random.randint(1, 85)),
            gen_string('latin1', random.randint(1, 255)),
            gen_string('numeric', random.randint(1, 255)),
            gen_string('utf8', random.randint(1, 85)),
        ]
```
looks better

**Comment by abalakh** [comment]
>tests should pass right away, without re-run. Appreciate if you can look on that more

@oshtaier fixed in SatelliteQE/robottelo-ci#928, please take a look

**Comment by odovzhenko** [comment]
Unit tests added.
```
/home/odovz/venv/sat-6.3.0/bin/python2.7 /home/odovz/.local/share/umake/ide/pycharm/helpers/pycharm/_jb_pytest_runner.py --path /home/odovz/projects/robottelo/tests/robottelo/test_datafactory.py -- -s
Console output is saving to: /home/odovz/robottelo.log
Launching py.test with arguments -s /home/odovz/projects/robottelo/tests/robottelo/test_datafactory.py in /home/odovz/projects/robottelo/tests/robottelo

============================= test session starts ==============================
platform linux2 -- Python 2.7.12, pytest-3.2.3, py-1.4.34, pluggy-0.4.0
rootdir: /home/odovz/projects/robottelo, inifile:
plugins: xdist-1.20.1, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collected 5 items
test_datafactory.py 

=========================== 5 passed in 3.14 seconds ===========================
```

**Pull request merged**